tempodb/encoding/appender.go

```diff
@@ -22,6 +23,7 @@ type appender struct {
 	records       map[uint64][]common.Record
 	hash          hash.Hash64
 	currentOffset uint64
+	mtx           sync.RWMutex
```

Comment:
So the "search headblock" is the only place where there is concurrent access between Append() and Records()

How about we separate the search headblock into a new field and lock that instead of adding the mtx to the appender?

Reply:
Added something that works to my personal branch - https://github.com/annanay25/tempo/commit/e2636556a7863f353ea25d89c6316c0926ad1ab9

Will push here if it looks good to you

Reply:
Pushed.